google/resource_compute_instance.go

```diff
@@ -715,6 +723,58 @@ func expandComputeInstance(project string, d *schema.ResourceData, config *Confi
 	}, nil
 }
 
+// return all possible Compute instances status except the one passed as parameter
+func getAllStatusBut(status string) []string {
```

Comment:
I'm not particulary proud of this function :
- I don't know if this is useful (it is used in `waitUntilInstanceHasDesiredStatus` `StateChangeConf` object)
- I can't find anywhere a list of possible Compute instance status so I must create the `allStatus` variable here (well, I could have created it in a `const` outside the function, even if it is the only usage here)
I am open to any suggestions to improve it.

Reply:
Seems fine to me. There isn't anywhere else that lists all the states, so we need it. One implementation you could compare to is what I did in GKE cluster recently: https://github.com/terraform-providers/terraform-provider-google/blob/master/google/resource_container_cluster.go#L1707-L1737 (+ [state utils](https://github.com/terraform-providers/terraform-provider-google/blob/master/google/state_util.go))

The goal there is to wait for _any_ stable ("resting") state, and not just the user's desired state.

Reply:
I've kept the old implementation because it's simpler to understand. Using `RestingStates` could have been confusing because here, as you said, there is no need to wait for a "resting" state, but rather just the user's desired state.